### ISSUE TITLE
Bearer token for HEAD operation with Filter is not applied

### ISSUE THREAD

**Issue by anatoly-bogatyrev** [nspcc-dev/neofs-node#215]
Bearer token for HEAD operation with Filter is not applied.
1) Public container
2) Created bearer token:

```
{
  "body": {
    "eaclTable": {
      "containerID": {
        "value": "EZqdi6vZXETAvKPxMLyjnb4muBLDCvEzABbmDPekhAk4"
      },
      "records": [
        {
          "operation": "GET",
          "action": "ALLOW",
          "filters": [
            {
              "headerType": "OBJECT",
              "matchType": "STRING_EQUAL",
              "key": "$Object:objectID",
              "value": "AW8PfM9EBmnap9HvLDvfUzkphZt3v2ctjXd5y595keng"
            }
          ],
          "targets": [
            {
              "role": "USER"
            }
          ]
        },
        {
          "operation": "PUT",
          "action": "ALLOW",
          "filters": [
            {
              "headerType": "OBJECT",
              "matchType": "STRING_EQUAL",
              "key": "$Object:objectID",
              "value": "AW8PfM9EBmnap9HvLDvfUzkphZt3v2ctjXd5y595keng"
            }
          ],
          "targets": [
            {
              "role": "USER"
            }
          ]
        },
        {
          "operation": "HEAD",
          "action": "ALLOW",
          "filters": [
            {
              "headerType": "OBJECT",
              "matchType": "STRING_EQUAL",
              "key": "$Object:objectID",
              "value": "AW8PfM9EBmnap9HvLDvfUzkphZt3v2ctjXd5y595keng"
            }
          ],
          "targets": [
            {
              "role": "USER"
            }
          ]
        },
        {
          "operation": "DELETE",
          "action": "ALLOW",
          "filters": [
            {
              "headerType": "OBJECT",
              "matchType": "STRING_EQUAL",
              "key": "$Object:objectID",
              "value": "AW8PfM9EBmnap9HvLDvfUzkphZt3v2ctjXd5y595keng"
            }
          ],
          "targets": [
            {
              "role": "USER"
            }
          ]
        },
        {
          "operation": "SEARCH",
          "action": "ALLOW",
          "targets": [
            {
              "role": "USER"
            }
          ]
        },
        {
          "operation": "GETRANGE",
          "action": "ALLOW",
          "filters": [
            {
              "headerType": "OBJECT",
              "matchType": "STRING_EQUAL",
              "key": "$Object:objectID",
              "value": "AW8PfM9EBmnap9HvLDvfUzkphZt3v2ctjXd5y595keng"
            }
          ],
          "targets": [
            {
              "role": "USER"
            }
          ]
        },
        {
          "operation": "GETRANGEHASH",
          "action": "ALLOW",
          "filters": [
            {
              "headerType": "OBJECT",
              "matchType": "STRING_EQUAL",
              "key": "$Object:objectID",
              "value": "AW8PfM9EBmnap9HvLDvfUzkphZt3v2ctjXd5y595keng"
            }
          ],
          "targets": [
            {
              "role": "USER"
            }
          ]
        },

    {
      "operation": "GET",
      "action": "DENY",
      "filters": [],
      "targets": [
        {
          "role": "USER",
          "keys": []
        }
      ]
    },
    {
      "operation": "HEAD",
      "action": "DENY",
      "filters": [],
      "targets": [
        {
          "role": "USER",
          "keys": []
        }
      ]
    },
    {
      "operation": "PUT",
      "action": "DENY",
      "filters": [],
      "targets": [
        {
          "role": "USER",
          "keys": []
        }
      ]
    },
    {
      "operation": "DELETE",
      "action": "DENY",
      "filters": [],
      "targets": [
        {
          "role": "USER",
          "keys": []
        }
      ]
    },
    {
      "operation": "SEARCH",
      "action": "DENY",
      "filters": [],
      "targets": [
        {
          "role": "USER",
          "keys": []
        }
      ]
    },
    {
      "operation": "GETRANGE",
      "action": "DENY",
      "filters": [],
      "targets": [
        {
          "role": "USER",
          "keys": []
        }
      ]
    },
    {
      "operation": "GETRANGEHASH",
      "action": "DENY",
      "filters": [],
      "targets": [
        {
          "role": "USER",
          "keys": []
        }
      ]
    }
      ]
    },
    "lifetime": {
      "exp": "100500",
      "nbf": "1",
      "iat": "0"
    }
  }
}
```

3) Try to HEAD object with allowed in the folder Object ID:
```
$ neofs-cli --rpc-endpoint s01.neofs.devenv:8080 --key KyidxRCWJXd34E9PteVnZFoGeSP8t9yipZ9tWJvHzUGJEjyS1xQ7 object head --cid EZqdi6vZXETAvKPxMLyjnb4muBLDCvEzABbmDPekhAk4 --oid AW8PfM9EBmnap9HvLDvfUzkphZt3v2ctjXd5y595keng --bearer bearer_allow_all_user 

Error: can't put object: could not send *object.HeadRequest: could not send head object request: rpc error: code = Unknown desc = access to operation 2 is denied by extended ACL check

```

4. The same issue is observed and for Delete operation, even in the case of the PUT-allowed operation without filters in the bearer token.

The same try to GET, GETRANGE has been executed as expected:
```
$ neofs-cli --rpc-endpoint s01.neofs.devenv:8080 --key KyidxRCWJXd34E9PteVnZFoGeSP8t9yipZ9tWJvHzUGJEjyS1xQ7 object get --cid EZqdi6vZXETAvKPxMLyjnb4muBLDCvEzABbmDPekhAk4 --oid AW8PfM9EBmnap9HvLDvfUzkphZt3v2ctjXd5y595keng --file local_file_eacl --bearer bearer_allow_all_user
[local_file_eacl] Object successfully saved
ID: AW8PfM9EBmnap9HvLDvfUzkphZt3v2ctjXd5y595keng
CID: EZqdi6vZXETAvKPxMLyjnb4muBLDCvEzABbmDPekhAk4
Owner: Na8a92h2dcBVPczpdYSXJy2P4xdnY7A83k
CreatedAt: 217
Size: 1024
HomoHash: 6a0d5e6d077cdfa77a6249139d56c79416fb6c0f8964ed8296f40b3e422885fb3539472a1f002d6b6415549ff1a7e57523b2ae499f23c9665054e8634c41a6e8
Checksum: f63d1a7bffbcf3fc17235c7fbaf546556570cd93a2d726edd7e98e350a4f4d55
Type: regular
Attributes:
  key1=1
  key2=abc
  FileName=da142f4d-7592-4ebf-a27c-aac62df28490
  Timestamp=1606350713 (2020-11-26 03:31:53 +0300 MSK)
```

**Comment by alexvanin** [comment]
#249 should fix this for head request. However there might be issues with getting split objects, but it is different issue.

**Comment by anatoly-bogatyrev** [comment]
Fixed by #249
